runtime/helpers.cu

```diff
@@ -106,6 +106,27 @@ __device__ float fmax(float a, float b) {
   }
 }
 
+__device__ __half fmax(__half a, __half b) {
+  return __half2float(a) > __half2float(b) ? a : b;
```

Comment:
`fmax(float, float)` also does NAN propagation. Don't we need to do something similar here as well?